types/merge-stream/index.d.ts

```diff
@@ -12,5 +12,6 @@ interface IMergedStream extends NodeJS.ReadWriteStream {
     isEmpty(): boolean;
 }
 
+declare function merge<T extends NodeJS.ReadableStream>(streams: T[]): IMergedStream;
```

Comment:
[From reading the source](https://github.com/grncdr/merge-stream/blob/1b33da64b219dcc8bf33645953c32a6fd9e3b36d/index.js#L21), I'm pretty sure the correct type is `(...streams: Array<T | T[]>)`

Reply:
An overload pair `{ (streams: T[]); (...streams: T[]) }` is actually not that common in Javascript because the single code path makes the `Array<T | T[]>` type *easier* to implement.

Reply:
First, sorry about taking so long to answer. I have been busy.
Indeed, your assessment is correct. The function transverses arrays recursively looking for streams. So whatever combination of streams and arrays of streams works.
However I believe the overload better exemplify the two forms you should use the function: passing streams individually or passing an array of streams.

Reply:
Good enough for me! Thanks for the explanation.